src/rust/engine/fs/src/lib.rs

```diff
@@ -683,6 +711,7 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
   }
 }
 
+#[derive(Debug, PartialEq)]
```

Comment:
The default `Debug` for this is going to be pretty noisy. Should probably override `Debug` to render only the path, and content length.

The `PartialEq` implementation seems potentially dangerous as well: if it's only being used for tests, should maybe add a helper there to compare content, or implement PartialEq only in that scope.

Reply:
What's dangerous about the `PartialEq`? That it's expensive?

Reply:
Yea. That it might end up getting used accidentally somewhere without considering the implications. Maybe not a big deal.

Reply:
Manually implemented Debug, manually implemented equality checking in the test